egs2/TEMPLATE/asr1/scripts/utils/evaluate_asr.sh

```diff
@@ -0,0 +1,266 @@
+    $0 --model_tag <model_tag> wav.scp asr_outputs
+
+    # Use pretrained model and perform inference and scoring
+    $0 --model_tag <model_tag> --stop-stage 2 --gt_text /path/to/text wav.scp asr_results
+
+    # Use local model and perform inference and scoring
+    $0 --asr_model_file /path/to/model.pth --stop-stage 2 --gt_text /path/to/text wav.scp asr_results
+
+EOF
+)
+
+log "$0 $*"
+# shellcheck disable=SC1091
+. utils/parse_options.sh
+
+wavscp=$1
+outdir=$2
+
+if [ $# -ne 2 ]; then
+    log "${help_message}"
+    exit 2
+fi
+
+# shellcheck disable=SC1091
+. ./path.sh
+# shellcheck disable=SC1091
+. ./cmd.sh
+
+# Check the option is valid
+if [ -z "${gt_text}" ] && [ "${stop_stage}" -ge 2 ]; then
+    log "--gt_text must be provided if perform scoring. Skipped scoring."
+    exit 1
+fi
+if [ -z "${model_tag}" ] && [ -z "${asr_model_file}" ]; then
+    log "Either --model_tag or --asr_model_file must be provided."
+    exit 1
+fi
+
+if ${gpu_inference}; then
+    # shellcheck disable=SC2154
+    _cmd="${cuda_cmd}"
+    _ngpu=1
+else
+    # shellcheck disable=SC2154
+    _cmd="${decode_cmd}"
+    _ngpu=0
+fi
+
+if [ ${stage} -le 1 ] && [ ${stop_stage} -ge 1 ]; then
+    _opts=()
+    if [ -n "${inference_config}" ]; then
+        _opts+=("--config")
+        _opts+=("${inference_config}")
+    fi
+    if [ -n "${asr_model_file}" ]; then
+        _opts+=("--asr_model_file")
+        _opts+=("${asr_model_file}")
+    fi
+    if [ -n "${lm_file}" ]; then
+        _opts+=("--lm_file")
+        _opts+=("${lm_file}")
+    fi
+    if [ -n "${model_tag}" ]; then
+        # FIXME: workaround until fixing filelock
+        espnet_model_zoo_download --unpack true "${model_tag}"
+        _opts+=("--model_tag")
+        _opts+=("${model_tag}")
+    fi
+
+    logdir="${outdir}/logdir"
+    mkdir -p "${logdir}"
+
+    # 1. Split the key file
+    key_file=${wavscp}
+    split_scps=""
+    _nj=$(min "${nj}" "$(wc -l < "${key_file}")")
+    for n in $(seq "${_nj}"); do
+        split_scps+=" ${logdir}/keys.${n}.scp"
+    done
+    # shellcheck disable=SC2086
+    utils/split_scp.pl "${key_file}" ${split_scps}
+
+    # 2. Submit decoding jobs
+    log "Decoding started... log: '${logdir}/asr_inference.*.log'"
+    # shellcheck disable=SC2086
+    ${_cmd} --gpu "${_ngpu}" JOB=1:"${_nj}" "${logdir}"/asr_inference.JOB.log \
+        python3 -m espnet2.bin.asr_inference \
+            --ngpu "${_ngpu}" \
+            --data_path_and_name_and_type "${wavscp},speech,sound" \
+            --key_file "${logdir}"/keys.JOB.scp \
+            --output_dir "${logdir}"/output.JOB \
+            "${_opts[@]}" ${inference_args}
+
+    # 3. Concatenates the output files from each jobs
+    for f in token token_int score gt_text; do
+        for i in $(seq "${_nj}"); do
+            cat "${logdir}/output.${i}/1best_recog/${f}"
+        done | LC_ALL=C sort -k1 >"${outdir}/${f}"
+    done
+fi
+
+# 4. Scoring
+if [ ${stage} -le 2 ] && [ ${stop_stage} -ge 2 ]; then
```

Comment:
This scoring part seems to be too simple and could not use `local/score.sh` (e.g., in swbd). 
But I think we should not make it complicated and this looks fine to me.
This is just a note.